sdk/python/storage/keeper_secrets_manager_storage/storage_oci_key_management/oracle_key_value_storage.py

```diff
@@ -0,0 +1,317 @@
+                    self.config = config or {}
+                    self.last_saved_config_hash = hashlib.md5(
+                        json.dumps(config, sort_keys=True, indent=4).encode()
+                    ).hexdigest()
+                except Exception as err:
+                    decryption_error = True
+                    self.logger.error(f"Failed to parse decrypted config file: {err}")
+                    raise Exception(f"Failed to parse decrypted config file {self.config_file_location}")
+
+            if json_error and decryption_error:
+                self.logger.info(f"Config file is not a valid JSON file: {json_error}")
+                raise Exception(f"{self.config_file_location} may contain JSON format problems")
+
+        except Exception as err:
+            self.logger.error(f"Error loading config: {err}")
+            raise err
+        
+    def change_key(self, new_key_id: str, new_key_version_id: str=None) -> bool:
+        old_key_id = self.key_id
+        old_key_version_id = self.key_version_id
+        old_crypto_client = self.crypto_client
+        old_management_client = self.management_client
+
+        try:
+            # Update the key and reinitialize the CryptographyClient
+            config = self.config
+            if not config:
+                self.load_config()
+            self.key_id = new_key_id
+            self.key_version_id = new_key_version_id
+            self.get_key_details()
+            self.__save_config({}, force=True)
+        except Exception as error:
+            # Restore the previous key and crypto client if the operation fails
+            self.key_id = old_key_id
+            self.key_version_id = old_key_version_id
+            self.crypto_client = old_crypto_client
+            self.management_client = old_management_client
+            self.get_key_details()
+            self.logger.error(
+                f"Failed to change the key to '{new_key_id}' for config '{self.config_file_location}': {error}"
+            )
+            raise Exception(f"Failed to change the key for {self.config_file_location}")
+
+        return True
+    
+    def get_key_details(self):
+        
+        opc_request_id = uuid.uuid4().hex.upper()
+        
+        key_details = self.management_client.get_key(key_id=self.key_id, opc_request_id=opc_request_id)
+        
+        algorithm = key_details.data.key_shape.algorithm
+        
+        if algorithm == KeyShape.ALGORITHM_RSA:
+            self.is_asymmetric = True
+        elif algorithm == KeyShape.ALGORITHM_AES:
+            self.is_asymmetric = False
+        else:
+            raise Exception(f"Unsupported key algorithm for the given key: {algorithm}")
+        
+        
+    
+    def read_storage(self) -> dict[str, str]:
+        if not self.config:
+            self.load_config()
+        return self.config
+    
+    def save_storage(self, updated_config: dict[str, str]) -> None:
+        self.__save_config(updated_config)
```

Comment:
If you are not doing anything else apart from calling the function then remove __save_config and keep just one function.